pkgs/by-name/ou/outputcheck/package.nix

```
@@ -0,0 +1,30 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "outputcheck";
+  version = "0.4.2";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "stp";
+    repo = "OutputCheck";
+    rev = "refs/tags/${version}";
+    hash = "sha256-QAlX+dmDKawnEzxtn9TToEmsA9jjzwO04l4bLM3fR/g=";
+  };
+
+  postPatch = ''
+    echo ${version} > RELEASE-VERSION
+  '';
+
```

You would do something like this: 
```suggestion
  checkPhase = ''
    # you put a shell script here which executes the tests
    # since this is very project specific there is no default thing I can provide you with
  '';
```

---

Done. However I have 1 failing test:
```
error: builder for '/nix/store/iiwqdkpx7m4lzwlgaaks0mbkghdf2nvs-outputcheck-0.4.2.drv' failed with exit code 1;
       last 10 log lines:
       > PASS: OutputCheck :: substitutions/simple-check.smt2 (51 of 52)
       > PASS: OutputCheck :: substitutions/simple-check-lit.smt2 (52 of 52)
       > ********************
       > Failed Tests (1):
       >   OutputCheck :: invalid-regex-syntax.smt2
       >
       >
       > Testing Time: 0.48s
       >   Passed: 51
       >   Failed:  1
       For full logs, run 'nix log /nix/store/iiwqdkpx7m4lzwlgaaks0mbkghdf2nvs-outputcheck-0.4.2.drv'.
```
Tried the nix log command but not much info why the test fail. Do you know a way how to debug this? Like nix develop or something.

---

you can do:
```nix
  checkPhase = ''
    lit -v tests/
  '';
```
then it prints out the error message.

---

Added an entry to postPatch to fix the failing test error. 
```
-- Testing: 52 tests, 8 workers --
PASS: OutputCheck :: check-next-first.smt2 (1 of 52)
PASS: OutputCheck :: check-not.smt2 (2 of 52)
PASS: OutputCheck :: cmdline_options/custom_check_prefix.smt2 (3 of 52)

...

PASS: OutputCheck :: invalid-regex-syntax.smt2 (27 of 52)

...

Testing Time: 0.47s
  Passed: 52
```